internal/apis/config/controller/v1alpha1/conversion.go

```diff
@@ -40,8 +41,8 @@ func Convert_controller_ControllerConfiguration_To_v1alpha1_ControllerConfigurat
 
 func Convert_Pointer_float32_To_float32(in **float32, out *float32, s conversion.Scope) error {
 	if *in == nil {
-		*out = 0
-		return nil
+		// We expect all nil pointers to have been defaulted to a non-nil value.
+		return fmt.Errorf("conversion from *float32(nil) -> float32 not supported")
```

Comment:
I think this is fine, but it doesn't follow the convention of the auto-generated conversion functions. Which means that we will throw errors for our defined types, but the ones that are covered in https://github.com/kubernetes/apimachinery/blob/5cb236977966cfc829e3b94b57fa69b44e0a95bc/pkg/apis/meta/v1/conversion.go will silently default.

Reply:
I reverted these changes.